From 9d15a6ea801be298c97ae131c77765923ff8a356 Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Sat, 7 Jan 2023 16:44:52 -0800
Subject: [PATCH 1/3] usb hal: Use vendor.typec.legacy property for legacy type
 c

- For some boards that doesn't have Type C PD option, use property to
fix the role and pd option.

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 usb/Usb.cpp | 6 ++++++
 usb/Usb.h   | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/usb/Usb.cpp b/usb/Usb.cpp
index c68d092..451c8ce 100644
--- a/usb/Usb.cpp
+++ b/usb/Usb.cpp
@@ -514,6 +514,12 @@ Status getTypeCPortNamesHelper(std::unordered_map<std::string, bool> *names) {
   DIR *dp;
   bool has_typec_port = false;
 
+  /* Enable Typ USB Legacy Support via vendor.typec.legacy property */
+  if (property_get_bool("vendor.typec.legacy", false)) {
+    ALOGE("Force Legacy device enabled");
+    return Status::ERROR;
+  }
+
   dp = opendir("/sys/class/typec");
   if (dp != NULL) {
     struct dirent *ep;
diff --git a/usb/Usb.h b/usb/Usb.h
index 0ffb568..299e3c3 100644
--- a/usb/Usb.h
+++ b/usb/Usb.h
@@ -24,6 +24,7 @@
 #include <android/hardware/usb/1.2/IUsbCallback.h>
 #include <android/hardware/usb/1.2/types.h>
 #include <android/hardware/usb/1.3/IUsb.h>
+#include <cutils/properties.h>
 #include <hidl/Status.h>
 #include <utils/Log.h>
 
@@ -102,4 +103,4 @@ struct Usb : public IUsb {
 }  // namespace hardware
 }  // namespace android
 
-#endif  // ANDROID_HARDWARE_USB_V1_3_USB_H
\ No newline at end of file
+#endif  // ANDROID_HARDWARE_USB_V1_3_USB_H
-- 
2.25.1

